### PR TITLE
refactor(ivy): refactoring cross_entropy loss

### PR DESCRIPTION
**Pull request by Infrared1029**
(WIP)

### CODE REVIEW

**Comment by a0m0rajab** [comment]
Hi @Infrared1029 since this is a WIP/Draft, if you like you can convert it through [GitHub Draft style](https://github.blog/2019-02-14-introducing-draft-pull-requests/), you can check this docs for [further details](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft).  

TLDR: you will find a convert to draft under the reviewers sidebar to use it: 
image

**Comment by Infrared1029** [comment]
> Hi @Infrared1029 since this is a WIP/Draft, if you like you can convert it through [GitHub Draft style](https://github.blog/2019-02-14-introducing-draft-pull-requests/), you can check this docs for [further details](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft).
> 
> TLDR: you will find a convert to draft under the reviewers sidebar to use it: image

Thanks @a0m0rajab, converted it into a nice draft.

**Comment by a0m0rajab** [comment]
Thanks for that!

**Comment by a0m0rajab** [comment]
Thanks!
